src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -215,6 +215,10 @@ export const useDiscoverHistogram = ({
    * Request params
    */
   const { query, filters } = useQuerySubscriber({ data: services.data });
+  const datasetFilters = useObservable(
+    stateContainer.internalState.state$,
+    stateContainer.internalState.getState()
+  ).datasetFilters;
```

Comment:
```suggestion
  const datasetFilters = useInternalStateSelector((state) => state.datasetFilters);
```
We have a state selector that we should be able to use for this.